src/GraphQL/ObjectExtensions.cs

```diff
@@ -169,6 +169,25 @@ string GetPropertyName(string key, out FieldType? field)
                     object? value = GetPropertyValue(item.Value, propertyInfo.PropertyType, field?.ResolvedType);
                     propertyInfo.SetValue(obj, value, null); //issue: this works even if propertyInfo is ValueType and value is null
                 }
+                else
+                {
+                    FieldInfo? fieldInfo;
+
+                    try
+                    {
+                        fieldInfo = type.GetField(propertyName, BindingFlags.IgnoreCase | BindingFlags.Public | BindingFlags.Instance);
```

Comment:
Only public fields? May be surprise for someone who overrided `GetRegisteredMembers`. But the same true for properties so maybe just add a note into docs.

Reply:
I plan to override ParseDictionary so this code isn't hit at all when using AutoRegisteringInputObjectGraphType.  I have not thought about exactly how this will be done yet.  However, this code will still be used for InputObjectGraphType.  As you can see, I added code for fields to match that of properties, which only supported public members.  Yes a comment is deserving.